libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -907,20 +907,24 @@ void AC_Fence::record_breach(uint8_t fence_type)
 void AC_Fence::record_margin_breach(uint8_t fence_type)
 {
     // if we haven't already breached a margin limit, update the breach time
+    const uint32_t now = AP_HAL::millis();
+
     if (!(_breached_fence_margins & fence_type)) {
-        const uint32_t now = AP_HAL::millis();
         _margin_breach_time = now;
-
-        // emit a message indicated we're newly-breached, but not too often
-        if (option_enabled(OPTIONS::MARGIN_BREACH)
-            && AP_HAL::timeout_expired(_last_margin_breach_notify_sent_ms, now, 1000U)) {
-            _last_margin_breach_notify_sent_ms = now;
-            print_fence_message("close", _breached_fence_margins | fence_type);
-        }
     }
 
     // update bitmask
     _breached_fence_margins |= fence_type;
+
+    // emit a message indicated we're newly-breached, but not too often
+    if (option_enabled(OPTIONS::NOTIFY_MARGIN_BREACH)
+        && AP_HAL::timeout_expired(_last_margin_breach_notify_sent_ms, now, 1000U)) {
+        _last_margin_breach_notify_sent_ms = now;
+        char msg[MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN+1];
+        ExpandingString e(msg, MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN+1);
```

Comment:
I've almost never seen this ExpandingString used.  @peterbarker you're OK with this?